### DISCUSSION TITLE
Asking for comments on my own understanding about using DP-GEN

### DISCUSSION THREAD

**Discussion by TerryHangLyu** [deepmodeling/dpgen#671] [Q&A] [answered]
Dear all,
     In most researches to train ML potential using deepMD-kit, the major way to obtain the data sets for training is to run AIMD to generate a long enough series of trajectories. I was about to apply this protocol in my own research. However, I then read the Zhihu article about DP-GEN. Compared with the mostly used but kind of 'brutal' protocol, seems like that DP-GEN can provide a more efficient way to sample the hypersurface because DP-GEN can proactively explore new regions of the hypersuface, which can not be well-described by the original trajectories.
     For example, if I want to train a ML potential to simulate a crysatllization process of an amorphous structure, normally, I need to run MD for both the amorphous phase and the crystalline phase. However, with DP-GEN, I can run the MD for the amorphous phase only and submit the trajectories of the amorphous phase to DP-GEN to have an initial roughly trained potential. And then, in the "sys_configs" part of the DP-GEN input file, I set the crystalline phase as the initial configuration of MD. And the temperature of the MD is set to be not very high so that the overall crystallinity can be maintained during the MD. As the MD goes on, the frame with high model deviation will be sample for first principle calculation. In this way, the original potential trained from amorphous trajectories will be modified and be capable of describing the crystal phase. In this way, with DP-GEN, the time to compute the crystalline trajectories using AIMD can be saved.
     This is my current understanding about using DP-GEN, but I am not sure about whether it is correct. Or it is still necessary to have the crystalline trajectories generated by AIMD. Could you please offer me comments on it?

Best wishes,
Hang LYU

**Comment by wanghan-iapcm** [comment, marked as answer]
Hello Hang, 

The key observation of ML potentials is that they cannot "extrapolate". They are not guaranteed to be accurate on the configurations that are not "covered" by the training dataset. In your case, you may want to sample both crystal and amorphous configurations, AND the configurations along the amorphous->crystal transition pathway. The sampling of the configuration by AIMD would be extremely expensive, that's why you may need DP-GEN, because the sampling is done by DPMD which is efficient, and the configurations that are not "covered" by the training dataset are selected from the DPMD trajectories according to model deviation. Sometimes, the time scale of the crystallization processes may be very long and is out of the capability of the DPMD simulations, then you may need to combine enhanced sampling techniques with DP to do the MD simulations. 

Best,
Han

**Reply by TerryHangLyu** [reply]
Thank you very much for your comment.
